src/main/java/org/alliancegenome/curation_api/jobs/executors/SequenceTargetingReagentExecutor.java

```diff
@@ -29,38 +35,107 @@ public void execLoad(BulkLoadFile bulkLoadFile) {
 			BulkFMSLoad fms = (BulkFMSLoad) bulkLoadFile.getBulkLoad();
 
 			SequenceTargetingReagentIngestFmsDTO sqtrIngestFmsDTO = mapper.readValue(
-					new GZIPInputStream(new FileInputStream(bulkLoadFile.getLocalFilePath())), SequenceTargetingReagentIngestFmsDTO.class);
+					new GZIPInputStream(new FileInputStream(bulkLoadFile.getLocalFilePath())),
+					SequenceTargetingReagentIngestFmsDTO.class);
 			bulkLoadFile.setRecordCount(sqtrIngestFmsDTO.getData().size());
 
-			AGRCurationSchemaVersion version = SequenceTargetingReagent.class.getAnnotation(AGRCurationSchemaVersion.class);
+			AGRCurationSchemaVersion version = SequenceTargetingReagent.class
+					.getAnnotation(AGRCurationSchemaVersion.class);
 			bulkLoadFile.setLinkMLSchemaVersion(version.max());
 
-			if (sqtrIngestFmsDTO.getMetaData() != null && StringUtils.isNotBlank(sqtrIngestFmsDTO.getMetaData().getRelease())) {
+			if (sqtrIngestFmsDTO.getMetaData() != null
+					&& StringUtils.isNotBlank(sqtrIngestFmsDTO.getMetaData().getRelease())) {
 				bulkLoadFile.setAllianceMemberReleaseVersion(sqtrIngestFmsDTO.getMetaData().getRelease());
 			}
 
 			BackendBulkDataProvider dataProvider = BackendBulkDataProvider.valueOf(fms.getFmsDataSubType());
 
 			List<Long> sqtrIdsLoaded = new ArrayList<>();
-			List<Long> sqtrIdsBefore = new ArrayList<>();
+			List<Long> sqtrIdsBefore = sqtrService.getIdsByDataProvider(dataProvider.name());
+			List<Long> sqtrGeneAssociationIdsLoaded = new ArrayList<>();
+			List<Long> sqtrGeneAssociationIdsBefore = sqtrService.getIdsByDataProvider(dataProvider.name());
```

Comment:
Isn't this getting the IDs of the SQTRs rather than the associations?

Reply:
I think we need to create association-specific DAO and Service classes that would be used for lookup of IDs by data provider, checking for existing associations, and handling cleanup.  The approach for the executor part looks good, but I think the rest of the stack should be a similar structure to the associations we have coming in from direct DQM loads (e.g. AlleleGeneAssociation).  You can think of it as two completely separate loads, just handled by the same executor.  First the SQTR load runs using classes in the SQTR stack, then the SQTR association load runs using classes from the SQTR association stack.

Reply:
@markquintontulloch Yeah, I wasn't sure if it made since break the association class into it's own service and validator and whatnot. I'll start working on that. Thanks!